docs/IR.md

```diff
@@ -1,127 +1,372 @@
 
-Names of tensor values flowing through the graph are unique - a particular name can reference either a graph input or an output of a single node. Reuse of names for several outputs is not allowed.
+The outputs of a given node introduce new names into the graph. The values of node outputs are computed by the node's operator. Node inputs MAY refer to node outputs, graph inputs, and graph initializers. When the name of a node output coincides with the name of a graph output, the graph output's value is the corresponding output value computed by that node.
+
+The graph must use single-static assignment for all node outputs. All node output namess MUST be unique within the graph.
+
+Node dependencies MUST NOT create cycles in the computation graph.
+
+The number of inputs and outputs in a node, their types, the set of attributes specified in a node and their types MUST satisfy the constraints imposed by the signature of the node’s operator.
+
+The list of nodes defining the top-level computation graph MUST be ordered topologically; that is, if node K follows node N in the graph, none of the data inputs of N may refer to outputs of K.
+
+Node attributes are used to pass literal (static) values to operators.
+
+#### Attributes
 
-The list of nodes defining the top-level computation graph MUST be ordered topologically \- that is, if node K follows node N in the graph, none of the data inputs of N may refer to outputs of K.
+Attributes, which represent literal constants passed to operators from nodes, have the following properties:
+
+Name|Type|Description
+|---|---|---|
+name|string|The name of the attribute. Must be unique among attributes, inputs, and outputs for any given operator and node.
+doc_string|string|A human-readable documentation for this value. Markdown is allowed.
+type|AttributeType|The type of the attribute, determining which of the remaining fields is used to hold the value of the attribute.
+f|float|A floating-point value.
+i|int64|An integer value
+S|byte[]|UTF-8 string
+t|Tensor|A tensor value
+g|Graph|A graph
+floats|float[]|List of floats
+ints|int64[]|List of integers
+strings|byte[][]|List of strings
+tensors|Tensor[]|List of tensor values
+graphs|Graph[]|List of graphs
+
+The properties ‘name’ and ‘type’ are required on all attributes, and ‘doc_string’ SHOULD be used on all attributes. An attribute MUST have only one of the value-carrying properties.
 
 #### Values
 
-The representation distinguishes between two kinds of values: attribute values, which are statically known, and runtime values. The type of values permitted in the two cases are different. The permitted types of attribute values are indicated by the enumeration `AttributeType`, while the permitted types of runtime values are described by `TypeProto`. 
+The representation distinguishes between two kinds of values: attribute values, which are statically known, and runtime values. The types of values permitted in the two cases are different.
```

Comment:
It'd be better to give some examples here. attribute values ---> attribute values (e.g., an operator's attributes); runtime values ---> runtime values (e.g., inputs and outputs of an operator). It's not extremely clear to me what runtime values are. As a standard, you might want this to be more specific.